src/routes/address/getTokensForContract.js

```diff
@@ -8,46 +8,35 @@ function getTokensForContract(req,res) {
 	let params = getParams(req);
 
 	// Get the defaults
-	let address  = params.address;
-	let contract = params.contract;
-	let limit    = Math.min(parseInt(params.limit, 10) || 50, 200);
-	let offset   = parseInt(params.offset, 10) || 0;
-
-	// Sanity checks
-	limit  = limit > 0   ? limit  : 50;
-	offset = offset >= 0 ? offset :  0;
-
+	const {address, contract, limit = 10, offset = 0}  = params;
+	
 	// Make sure that the email doesn't exist already
 	Database.connect((Client) => {
-		Client.query(AddressQueries.getTokensForContract(address, contract, limit, offset), (result) => {
+		let total = 0;
+		Client.query(AddressQueries.getTokensForContractAmount(address, contract)).then(result => total = Number(result.rows[0].count));
```

Comment:
You can use result.rowCount, which is Postgres' native row count return value

Reply:
But I've created the query to make it count. You can check on the `AddressesQueries` on line 180. It's a query only to return the count of rows instead of fetching the data itself and then counting the number of rows. Does it make sense to remove this `count` and make the complete query to get the `rowCount`?

We can use the `getTokensForContract` because it has the Limit and Offset, so if I try to count, the amount will return only the value inside the limit and offset. We need the total amount without those filters for the pagination on the front end.

Reply:
You're right -- cleaned up the query to simplify complexity, marking this as resolved.